pkg/geo/geo.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+// Package geo contains types that represent spatial objects on a plane or
+// sphere. These respectively correspond to the SQL GEOMETRY and GEOGRAPHY
+// types.
+//
+// Subpackages are available that perform operations using these types:
+// - geo/geofn implements the logic of the PostGIS/OGC functions over GEOMETRY
+//   and GEOGRAPHY types.
+// - geo/geoindex implements index-accelerated versions of a small number of the
+//   functions in geofn. These are used with a precomputed SQL inverted index.
+package geo
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/cockroachdb/errors"
+	"github.com/golang/geo/s2"
+)
+
+// Region represents a geometrical object in planar or spherical space.
+//
+// WIP This needs a better name. GeoJSON calls it a "Geometry", but that's
+// confusing with the SQL GEOMETRY type. Does PostGIS have a name for this? A
+// casual scan seems to show that they totally separate the two. Now that I
+// think about it, maybe we actually do need two types, on that wraps `r2` and
+// one that wraps `s2`.
+type Region struct {
+	// WIP The prototype has made it pretty clear that using s2.Region here
+	// results in awkward type switches all over the place. There's got to be
+	// something better for us to do here.
+	s2.Region
```

Comment:
wondering if we should make `s2.Region` the thing to inherit, or to make our own struct hierarchy that implements the inheritance properly. using `oneof` a la protobufs comes to mind....